File: go/2706.go
package main

func buyChoco(prices []int, money int) int {
	var a, b int
	if prices[0] < prices[1] {
		a, b = prices[0], prices[1]
	} else {
		a, b = prices[1], prices[0]
	}
	for i := 2; i < len(prices); i++ {
		if prices[i] < b {
			if prices[i] < a {
				a, b = prices[i], a
			} else {
				b = prices[i]
			}
		}
	}
	if a+b > money {
		return money
	}
	return money - a - b
}

File: python/2706.py
from typing import List

class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        a, b = prices[0], prices[1]
        for c in prices[2:]:
            if c < a:
                if a < b:
                    b = a
                a = c
            elif c < b:
                b = c
        return money if money - a - b < 0 else money - a - b
